src/pages/Swap/index.tsx

```diff
@@ -526,42 +524,16 @@ export function Swap({
   const priceImpactTooHigh = priceImpactSeverity > 3 && !isExpertMode
   const showPriceImpactWarning = largerPriceImpact && priceImpactSeverity > 3
 
-  // Handle time based logging events and event properties.
+  const prevTrade = usePrevious(trade)
   useEffect(() => {
-    const now = new Date()
-    // If a trade exists, and we need to log the receipt of this new swap quote:
-    if (newSwapQuoteNeedsLogging && !!trade) {
-      // Set the current datetime as the time of receipt of latest swap quote.
-      setSwapQuoteReceivedDate(now)
-      // Log swap quote.
-      sendAnalyticsEvent(SwapEventName.SWAP_QUOTE_RECEIVED, {
-        ...formatSwapQuoteReceivedEventProperties(
-          trade,
-          trade.gasUseEstimateUSD ?? undefined,
-          fetchingSwapQuoteStartTime
-        ),
-        ...trace,
-      })
-      // Latest swap quote has just been logged, so we don't need to log the current trade anymore
-      // unless user inputs change again and a new trade is in the process of being generated.
-      setNewSwapQuoteNeedsLogging(false)
-      // New quote is not being fetched, so set start time of quote fetch to undefined.
-      setFetchingSwapQuoteStartTime(undefined)
-    }
-    // If another swap quote is being loaded based on changed user inputs:
-    if (routeIsLoading) {
-      setNewSwapQuoteNeedsLogging(true)
-      if (!fetchingSwapQuoteStartTime) setFetchingSwapQuoteStartTime(now)
-    }
-  }, [
-    newSwapQuoteNeedsLogging,
-    routeIsSyncing,
-    routeIsLoading,
-    fetchingSwapQuoteStartTime,
-    trade,
-    setSwapQuoteReceivedDate,
-    trace,
-  ])
+    if (!trade || prevTrade === trade) return // no new swap quote to log
+
+    setSwapQuoteReceivedDate(new Date())
```

Comment:
this property is still used downstream in child components, so not getting rid of it for now